### PR TITLE
Run Agones tests in CI

### PR DESCRIPTION
**Pull request by markmandel**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug
> /kind cleanup
> /kind documentation

/kind feature

> /kind hotfix

**What this PR does / Why we need it**:

This makes it such that on each PR, the Agones integration tests are run
against the GKE + Agones cluster specified in ./agones/main.tf

This includes:
* Fix for main.tf to align release branch with Agones version.
* Configure how long to wait before cleaning up past test namespaces.
* Cleanup and optimisation of cloudbuild.yaml.

**Which issue(s) this PR fixes**:

Work on #510

**Special notes for your reviewer**:

### CODE REVIEW

File: cloudbuild.yaml [hunk 1]
```diff
@@ -16,35 +16,34 @@ steps:
   - name: gcr.io/cloud-builders/git
     args: [ submodule, update, --init, --recursive ]
     id: fetch-git-submodules
+    waitFor:
+      - "-"
```
Comment (XAMPPRocky) [inline cloudbuild.yaml:20]: Maybe add a comment here explaining this. Maybe this is obvious to people who use cloudbuild, but it's not obvious to me compared to other usages of waitFor.

File: agones/src/lib.rs [hunk 2]
```diff
@@ -75,10 +78,23 @@ async fn setup_namespace(client: kube::Client) -> String {
     let lp = ListParams::default().labels("owner=quilkin-test");
     let nss = namespaces.list(&lp).await.unwrap();
     let dp = DeleteParams::default();
+
+    let delay = env::var_os(DELETE_DELAY_SECONDS)
+        .map(|value| chrono::Duration::seconds(value.into_string().unwrap().parse().unwrap()));
```
Comment (XAMPPRocky) [inline agones/src/lib.rs:83]: Nit: You don't need to use `var_os` if you're converting to string anyway, and you can flatten the unwraps.

```suggestion
    let delay = env::var(DELETE_DELAY_SECONDS)
        .as_deref()
        .ok()
        .and_then(|s| i64::parse(s).ok())
        .map(chrono::Duration::seconds);
```

**Review by XAMPPRocky — APPROVED** [review]
LGTM, only nits

File: cloudbuild.yaml (hunk 1 above)
Comment (markmandel) [reply cloudbuild.yaml:20]: Excellent suggestion - will do.

**Comment by markmandel** [comment]
GCP IAM permission on fail (which I expected but wasn't sure if we would hit).

File: agones/src/lib.rs (hunk 2 above)
Comment (markmandel) [reply agones/src/lib.rs:83]: I swear I never saw `env::var()` existing before. Totally missed that. Thank you!

**Comment by markmandel** [comment]
That took an hour - I should look at why. I think some of my optimisations in cloudbuild.yaml caused a bunch of redundant work to be done.

**Comment by markmandel** [comment]
Ah! Found it.

**Pull request merged**